src/wasm-stack.h

```diff
@@ -232,7 +232,9 @@ void BinaryenIRWriter<SubType>::visitPossibleBlockContents(Expression* curr) {
 
 template<typename SubType>
 void BinaryenIRWriter<SubType>::visit(Expression* curr) {
-  emitDebugLocation(curr);
+  if (!curr->is<Call>()) {
+    emitDebugLocation(curr);
+  }
```

Comment:
Please add a comment explaining why `Call` is special here. That's not obvious to me, sorry if I didn't understand something you already explained in the issue.

In particular, why is `Call` different from, say, `Binary`? And what about `CallIndirect` and `CallRef`?